test/run_tests.py

```diff
@@ -42,20 +42,87 @@ class Result(Enum):
     SKIPPED = 3
 
 
-def try_decode(binary_object):
+class Completion:
```

Comment:
Looks like type hinting is forgotten here

Reply:
I explicitly typed `completed` in `__init__`, and left `lock` and `condition` to be implicitly typed based on their initialization

Reply:
not sure if I understand. I was all methods signatures - id/success are unhinted.

Reply:
Oh youre so right i missed those, thanks! I misunderstood originally